apps/console/src/features/authentication-flow-builder/components/script-editor-panel/adaptive-script-reset-confirmation-modal.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { Code, ConfirmationModal, ConfirmationModalPropsInterface } from "@wso2is/react-components";
+import React, { FunctionComponent, MouseEvent, ReactElement } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import useAuthenticationFlow from "../../hooks/use-authentication-flow";
+
+/**
+ * Proptypes for the adaptive script reset confirmation modal component.
+ */
+export type AdaptiveScriptResetConfirmationModalPropsInterface = Partial<ConfirmationModalPropsInterface> & 
+    IdentifiableComponentInterface
+
+/**
+ * Adaptive script reset confirmation modal.
+ *
+ * @param props - React component props.
+ * @returns Adaptive script reset confirmation modal.
+ */
+const AdaptiveScriptResetConfirmationModal: FunctionComponent<AdaptiveScriptResetConfirmationModalPropsInterface> = (
+    props: AdaptiveScriptResetConfirmationModalPropsInterface
+): ReactElement => {
+    const { open, onClose, ["data-componentid"]: componentId, ...rest } = props;
+
+    const { t } = useTranslation();
+
+    const {
+        onConditionalAuthenticationToggle
+    } = useAuthenticationFlow();
+
+    /**
+     * This will reset the adaptive script to the default one
+     * when the user gives their consent for reseting script editor.
+     */
+    const resetAdaptiveScriptTemplateToDefaultHandler = () => {
+        onConditionalAuthenticationToggle(false);
+        onClose(null, null);
```

Comment:
If onclose works without parameters passed, let's mark its parameters as optional and avoid passing null here.